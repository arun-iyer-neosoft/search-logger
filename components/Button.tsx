import React from "react";

interface ButtonProps {
	type?: "button" | "submit" | "reset";
	color?: "blue" | "gray";
}

export const Button: React.FC<
	ButtonProps & React.HTMLProps<HTMLButtonElement>
> = ({ children, type = "button", color = "blue", ...rest }) => {
	const colorClass =
		color === "blue"
			? " bg-blue-700 border-blue-800 text-white hover:bg-blue-800"
			: " bg-gray-200 border-gray-300 hover:bg-gray-300";
	return (
		<button
			className={
				"px-4 py-2 min-w-[120px] text-cente border self-end" + colorClass
			}
			{...rest}
			type={type}
		>
			{children}
		</button>
	);
};
