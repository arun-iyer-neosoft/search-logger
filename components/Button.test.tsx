import { Button } from "./Button";
import { screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { configure, render, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Button", () => {
	it("should have blue background with bg-blue-700 class", () => {
		const wrapper = shallow(<Button>Test</Button>);

		expect(wrapper.find("button").hasClass("bg-blue-700")).toBe(true);
	});

	it("should have gray background with bg-gray-200 class", () => {
		const wrapper = shallow(<Button color='gray'>Test</Button>);

		expect(wrapper.hasClass("bg-gray-200"));
	});

	it("should match snapshot on hover", () => {
		const wrapper = shallow(<Button>Test</Button>);

		wrapper.simulate("ommouseenter");
		expect(wrapper).toMatchSnapshot();

		wrapper.simulate("onmouseleave");
		expect(wrapper).toMatchSnapshot();
	});

	it("should have type button", () => {
		const wrapper = shallow(<Button>Test</Button>);

		expect(wrapper.type()).toBe("button");
	});

	// it("should have gray background with bg-gray-200 class", () => {
	// 	render(<Button color='gray'>Test</Button>);

	// 	const buttonWithClass = screen.getByText("Test");

	// 	expect(buttonWithClass).toHaveClass("bg-gray-200");
	// });
});
