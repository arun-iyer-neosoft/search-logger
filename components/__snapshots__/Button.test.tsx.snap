// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot on hover 1`] = `ShallowWrapper {}`;

exports[`Button should match snapshot on hover 2`] = `ShallowWrapper {}`;
